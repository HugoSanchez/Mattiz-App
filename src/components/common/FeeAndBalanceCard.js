import React from 'react';
import { View, Text } from 'react-native';

import styles from '../../styles';

// Props:
// @ fee:           - Required. Fee amount to display.
// @ balance:       - Required. Balance amount to display.

const FeeAndBalanceCard = props => {

    const { 
        container,
        feeBox,
        balanceBox,
        balanceTitle
    } = compStyles;

    const {
        extraSmallLightTitle,
        extraSmallRegularNumber
    } = styles;


    console.log(props.fee)
    console.log(props.balance)
    
    return (
        <View style={ container }>
            <View style={ feeBox }>
                <Text style={ extraSmallLightTitle }>
                    Fee:  
                        <Text style={ extraSmallRegularNumber }>
                            { '   $' + props.fee }
                        </Text>
                </Text>
            </View>

            <View style={ balanceBox }>
                <Text style={ balanceTitle }>
                    Balance:  
                        <Text style={ extraSmallRegularNumber }>
                        { '   $' + props.balance }
                    </Text>
                </Text>
            </View>
        </View>
    );
}

const compStyles = {
    container: { 
        flex: 2, 
        flexDirection: 'row', 
        marginLeft: '7%',
        marginRight: '7%'
    },
    feeBox: { 
        flex: 1, 
        flexDirection: 'row',  
        alignItems: 'center',
        justifyContent: 'center'
    },
    balanceBox: { 
        flex: 1.5, 
        flexDirection: 'row',  
        alignItems: 'center',
        justifyContent: 'center',
        marginRight: '4%'
    },
    balanceTitle: [ 
        styles.extraSmallLightTitle, { 
            marginLeft: 25 
        } 
    ]
}

export default FeeAndBalanceCard;