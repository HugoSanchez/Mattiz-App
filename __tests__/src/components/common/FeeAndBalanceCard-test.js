import 'react-native'
import React from 'react'
import FeeAndBalanceCard from '../../../../src/components/common/FeeAndBalanceCard'
import {render, fireEvent} from '@testing-library/react-native'

test('Renders with title', () => {
	const {container} = render(<FeeAndBalanceCard />)
	expect(container.children[0]).toMatchSnapshot()
})
