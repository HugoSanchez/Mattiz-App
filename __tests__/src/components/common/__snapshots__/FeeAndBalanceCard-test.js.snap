// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with title 1`] = `
<View
  style={
    Object {
      "flex": 2,
      "flexDirection": "row",
      "marginLeft": "7%",
      "marginRight": "7%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#154053",
          "fontFamily": "Rajdhani-Light",
          "fontSize": 18,
        }
      }
    >
      Fee:
      <Text
        style={
          Object {
            "color": "#154053",
            "fontFamily": "Rajdhani-Regular",
            "fontSize": 16,
          }
        }
      >
           $undefined
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1.5,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": "4%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#154053",
            "fontFamily": "Rajdhani-Light",
            "fontSize": 18,
          },
          Object {
            "marginLeft": 25,
          },
        ]
      }
    >
      Balance:
      <Text
        style={
          Object {
            "color": "#154053",
            "fontFamily": "Rajdhani-Regular",
            "fontSize": 16,
          }
        }
      >
           $undefined
      </Text>
    </Text>
  </View>
</View>
`;
